core/src/main/java/org/apache/iceberg/io/FanoutWriter.java

```diff
@@ -36,7 +36,7 @@
  * writer may potentially consume substantially more memory compared to {@link ClusteredWriter}. Use
  * this writer only when clustering by spec/partition is not possible (e.g. streaming).
  */
-abstract class FanoutWriter<T, R> implements PartitioningWriter<T, R> {
+public abstract class FanoutWriter<T, R> implements PartitioningWriter<T, R> {
```

Comment:
It was actually done on purpose as it is a parent class. We expose `FanoutDataWriter` and related implementations. The same is also true for `ClusteredWriter`. Is there a particular need to expose this class? Is it something generic enough to live in the repo intself?

Reply:
The only difference between `FanoutWriter` and `FanoutDataWriter` is that the latter enforces the use of `RollingDataWriter` and adds quite a bit of complexity. I'd argue that both are equally useful and `FanoutWriter` is much easier to implement and does not enforce things like file name format for example.